File: Keystroke Problem/keystroke.py
#Python Program to find maximum no of A's
def optimal(n):
    if n<=6:
        return n
    maxi = 0
    for b in range(n-3,0,-1):
        curr = (n-b-1)*optimal(b)
        if curr > maxi:
            maxi = curr
    return maxi

n = int(input("Enter no of keystrokes :- "))
print("Maximum no with ",n,"keystrokes is ",optimal(n))

File: Keystroke Problem/keystroke.c
//C Program to find maximum no of A's
#include <stdio.h>
static int optimal(int n)
{
    if(n<=6)
    {
        return n;
    }
    int max = 0,b;
    for(b = n-3;b>=1;b--)
    {
        int curr = (n-b-1)*optimal(b);
        if(curr > max)
        {
            max = curr;
        }
    }
    return max;
}
int main()
{
    int n = 7;
    printf("Maximum no with %d keystrokes is %d",n,optimal(n));
}
